processing/src/main/java/org/apache/druid/segment/data/ObjectStrategy.java

```diff
@@ -77,6 +78,18 @@ default boolean readRetainsBufferReference()
     return true;
   }
 
+  /**
+   * Preferred byte order to read values from a buffer with {@link #fromByteBuffer(ByteBuffer, int)} and similar methods
```

Comment:
Should also update `fromByteBuffer` contract to say that the caller must pass in a buffer whose order matches `getByteOrder()`. Please double-check all callers of `fromByteBuffer` to make sure they do that properly.